src/services/items/interfaces/item.ts

```diff
@@ -9,15 +19,61 @@ export interface ItemSettings extends Serializable {
   isCollapsible?: boolean;
 }
 
-export interface Item<T extends UnknownExtra = UnknownExtra, S = ItemSettings> {
+export type ItemBase<S = ItemSettings> = {
   id: string;
   name: string;
   description: string;
-  type: ItemType;
   path: string;
-  extra: T;
   settings: S;
   creator: string;
   createdAt: string;
   updatedAt: string;
-}
+};
+
+export type AppItemType<S = ItemSettings> = {
+  type: `${ItemType.APP}`;
+  extra: AppItemExtra;
+} & ItemBase<S>;
```

Comment:
You'll have much much much better review from @dialexo  
but here would it make sense to `extends ItemBase` and have an interface?

Reply:
I think @spaenleh wrote a type alias instead of an interface so that he could use an union of the types (and so the compiler can infer a discriminated union between the different `ItemType`s). I guess that's one of those TS quirks where you can write almost the same thing in many different ways, but some features require specific syntax :shrug: